### ISSUE TITLE
[Feature Request] Counter for Events

### ISSUE THREAD

**Issue by Schrolli91** [variar/klogg#615]
In an internal LogViewer of my Company we have a feature so called counter to count events inside the log files.
This will be done by setting a Regex for UP-counting and one for DOWN-counting.
![image](https://github.com/variar/klogg/assets/6043382/33a8ca9b-ea10-45f5-ba8a-d9355731f4f6)

This makes it rely easy to detect memory leaks for example (more allocs then frees).
![image](https://github.com/variar/klogg/assets/6043382/60776484-b2fc-4045-8d65-26046ffbec12)

Maybe such an counter will be a useful tool for others.

**Comment by variar** [comment]
I see this being useful in some cases. I am not sure what the UI for this feature in klogg should be. Klogg shows a count of matches, so doing two searches and subtracting the matches counts achieves the same result, but indeed in a more cumbersome manner.

**Comment by Schrolli91** [comment]
The idea in our tool is that I can see the count at any point in time.

For example, if our application is dynamically allocating and freeing buffers for received data, I can see the current number of allocated memory pools at any time (regex on 
 `sendBufferAlloc`/`sendBufferFree`).

This is not possible if only a "total difference" between two searches is shown.

**Comment by Schrolli91** [comment]
Maybe such an function can be integrated via an "plugin"-interface or something?

**Comment by variar** [comment]
Yes, this seems to be a good use case for plugins. There was a fork of klogg that implemented python plugins support. Not sure where it is now.

**Comment by Schrolli91** [comment]
Unfortunately i had no luck to find the fork.

**Comment by variar** [comment]
Fork is here https://github.com/dkinsbur/klogg. However, it is quite outdated
